app/models/veteran.rb

```diff
@@ -37,6 +37,8 @@ class Veteran < CaseflowRecord
 
   delegate :full_address, to: :address
 
+  before_save :set_death_reported_at!
```

Comment:
IMO, it's easier to determine relevance of this hook if you check `if date_of_death_changed?` here.
```suggestion
  before_save : set_death_reported_at! if :date_of_death_changed?
```
or maybe `if :will_save_change_to_date_of_death?` https://stackoverflow.com/questions/24314584/run-a-callback-only-if-an-attribute-has-changed-in-rails

Reply:
Good optimization! Updated.